### PR TITLE
BING-59: create watchlist add functionality

### PR DESCRIPTION
**Pull request by MatasRut**

### CODE REVIEW

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistsControler.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        Object.keys(watchlists).length !== 0 &&
+        watchlists.constructor !== Object
+      )
+        setWatchlists(watchlists);
+    };
+    loadWatchlists();
+  }, []);
+
+  const handleCreateWatchlist = async (name) => {
+    const newWatchlist = await api.createWatchlist({ name: name });
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistsControler.jsx]: The api call can return `undefined` which results in an unhandled exception if unchecked

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistsControler.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from 'react';
+import { WatchlistAdd } from 'components/WatchlistsControler/WatchlistAdd/WatchlistAdd';
+import { WatchlistPicker } from 'components/WatchlistsControler/WatchlistPicker/WatchlistPicker';
+import { List } from '@material-ui/core';
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistsControler.jsx]: This is unused

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx [hunk 3]
```diff
@@ -5,30 +5,18 @@ import MovieIcon from '@material-ui/icons/Movie';
 import ListItemText from '@material-ui/core/ListItemText';
 import { withRouter } from 'react-router-dom';
 import { ExpandLess, ExpandMore } from '@material-ui/icons';
-import 'components/Watchlist/WatchlistPicker/WatchlistPicker.scss';
+import './WatchlistPicker.scss';
+import Typography from '@material-ui/core/Typography';
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx]: There are a few named and default exports intermixed from `@material-ui/core` and `utils.api` is unused

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx [hunk 4]
```diff
@@ -43,22 +31,33 @@ export const WatchlistPicker = withRouter(
...
+                className={
+                  history.location.pathname.includes(watchlist._id)
+                    ? 'item-text selected'
+                    : 'item-text'
+                }
+                onClick={() => {
+                  history.push(`/watchlists/${watchlist._id}`);
+                }}
+              >
+                {watchlist.name}
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx]: I think using `ListItemText` would look nicer, the `inset` prop gives a quick and easy way to align them for example
```suggestion
                <ListItemText
                    primary={watchlist.name}
                    disableTypography
                    inset
                    className='item-text'
                  />
```

the `disableTypography` is needed here to achieve the overflow-ellipses and selected highlight

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  ListItemIcon,
+  ListItemText,
+  TextField,
+} from '@material-ui/core';
+import Slide from '@material-ui/core/Slide';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogUI from '@material-ui/core/Dialog';
+import './WatchlistAdd.scss';
+import AddCircleIcon from '@material-ui/icons/AddCircle';
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx]: It would be best to import everything from `@material-ui/core` with named exports. Also, separating local imports could reduce possible confusion

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction='up' ref={ref} {...props} />;
+});
+
+export function WatchlistAdd({ onCreateWatchlist, openDrawer }) {
+  const [dialogOpen, setDialogOpen] = useState(false);
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx]: I'd prefer to name this as `isDialogOpen`, it already is quite confusing with `closeDialog` being a handler but named similarly

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+  return (
+    <>
+      <ListItem button onClick={openDialog} key={'create'}>
+        <ListItemIcon>
+          <AddCircleIcon />
+        </ListItemIcon>
+        <ListItemText primary={'New Watchlist'} />
+      </ListItem>
+      <DialogUI
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx]: I think this should be moved to a separate file as it's own component. It's already rather large and could potentially grow in the future

**Review by palianskas — CHANGES_REQUESTED** [review]
Overall there's mostly minor things like import management. I believe the watchlist list's items could have a better layout. The main thing is that the api call to create a watchlist might fail but it wouldn't be handled right now

P.S. The drawer  _Watchlist_ list item could be renamed via this too

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx (hunk 4 above)
Comment (MatasRut) [reply app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx]: I prefer the other way, there are many places where we use Typography and this barely saves any code

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx (hunk 7 above)
Comment (MatasRut) [reply app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx]: Created a separate Basic Dialog component that should be used where a basic dialog is needed. This reduces the code quite a bit so I think this can stay like this

File: app/bingey-ui/src/components/App/App.jsx [hunk 8]
```diff
@@ -11,7 +11,7 @@ export const App = () => {
     <>
       <Router>
         <Switch>
-          <Route exact path='/admin'>
+          <Route path='/admin'>
```
Comment (palianskas) [inline app/bingey-ui/src/components/App/App.jsx:14]: This change doesn't seem related, why is this here?

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import {
+  Collapse,
+  ListItemIcon,
+  ListItem,
+  ListItemText,
+} from '@material-ui/core';
+import MovieIcon from '@material-ui/icons/Movie';
+import { ExpandLess, ExpandMore } from '@material-ui/icons';
+import './WatchlistPicker.scss';
+import Typography from '@material-ui/core/Typography';
```
Comment (palianskas) [inline app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx]: This should go under the named imports

File: app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx (hunk 4 above)
Comment (palianskas) [reply app/bingey-ui/src/components/WatchlistsControler/WatchlistPicker/WatchlistPicker.jsx]: As it is now, the watchlists are hierarchically on the same indentation as the drawer's _'Watchlists' folder_ item which makes them seem out of place. The `inset` prop on `ListItemText` solves this easily

_edit_ confused PRs and though that watchlist rendering was added via this. This change request should be ignored

**Review by palianskas — CHANGES_REQUESTED** [review]
I think the _WatchlistController_ directory could be moved under _Navigation/Drawer_ as it is part of it

File: app/bingey-ui/src/components/App/App.jsx (hunk 8 above)
Comment (MatasRut) [reply app/bingey-ui/src/components/App/App.jsx:14]: My bad, I forgot to comment on this. Found a bug where refreshing on certain admin tabs would not render the admin panel but the main page, this fixes it.

**Review by palianskas — APPROVED** [review]
